### PR TITLE
Set min version of cargo-make for diplomat-get-rev task

### LINKED ISSUE #1050: `cargo make diplomat-get-rev` fails for me locally
```
➤ cargo make diplomat-get-rev
[cargo-make] INFO - cargo make 0.32.10
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: diplomat-get-rev
[cargo-make] INFO - Profile: development
[cargo-make] INFO - Running Task: diplomat-get-rev
[cargo-make] ERROR - Error while running duckscript: Source: Unknown Line: 6 - invalid number at line 1 column 2
[cargo-make] WARN - Build Failed.
```

It fails for me on this line: `metadata = json_parse --collection ${metadata}`.

This command is required for the diplomat install. I worked around it by modifying the command:

```
[tasks.diplomat-get-rev]
description = "Get current Diplomat revision"
category = "ICU4X Development"
script_runner = "@duckscript"
script = '''
echo d98d31bdbdad3af98493ccfdc955b32fca886176
'''
```

I checked the output of `cargo metadata` by running `cargo metadata > metadata.json` and then parsing the JSON in node. The JSON appears well formed. It is quite large though at `1.1M`. I don't know if that is making the duckscript fail due to the size of the file.

### PR DESCRIPTION
**Pull request by echeran**
Fixes #1050 .

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Review by gregtatum — APPROVED** [review]
Thanks!

**Comment by echeran** [comment]
@Manishearth Do you know why this change would cause the WASM build-test job in CI to fail?

**Comment by Manishearth** [comment]
Looks like your "note" got included in the `diplomat-get-rev` output

**Comment by Manishearth** [comment]
Print to stderr perhaps, or handle this elsewhere, the output of get-rev has to be a single hash

File: tools/scripts/ffi.toml [hunk 1]
```diff
@@ -147,6 +147,7 @@ args = ["build", "--package", "icu", "--target", "thumbv7m-none-eabi"]
 [tasks.diplomat-get-rev]
 description = "Get current Diplomat revision"
 category = "ICU4X Development"
+install_crate = { crate_name = "cargo-make", binary = "cargo", test_arg = ["make", "--version"], min_version = "0.35.0" }
```
Comment (Manishearth) [inline tools/scripts/ffi.toml:150]: should this instead be a separate task that a couple tasks may depend on

**Comment by sffc** [comment]
What if `diplomat-get-rev` spawned a new session of `cargo make`?  Would that eliminate the condition where the first time running `cargo make diplomat-get-rev` fails, and the second time succeeds?

**Review by sffc — COMMENTED** [review]
(setting review bit)

File: tools/scripts/ffi.toml (hunk 1 above)
Comment (echeran) [reply tools/scripts/ffi.toml]: Done. Yep, agree with both your and Shane's reasonings.

**Review by Manishearth — APPROVED** [review]

**Review by sffc — DISMISSED** [review]

**Comment by echeran** [comment]
The latest push just now looks good and is intended to be extra sure to avoid the CI failure. I couldn't totally understand the [log from the previous commit's CI failure](https://github.com/unicode-org/icu4x/runs/3676646195?check_suite_focus=true), but I thought it may be related to changing the console output when it executes `cargo make --version`, so I cut that out with the newly pushed commit.

Locally, I verified with the following:

Down-rev'ing `cargo-make`:
```
$ cargo install --version 0.33.0 --force cargo-make
$ cargo make --version
cargo-make 0.33.0
```
Running the first time:
```
$ cargo make diplomat-get-rev
[cargo-make] INFO - cargo make 0.33.0
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: diplomat-get-rev
[cargo-make] INFO - Profile: development
[cargo-make] INFO - Execute Command: "cargo" "install" "--force" "cargo-make"
    Updating crates.io index
  Installing cargo-make v0.35.0
   Compiling libc v0.2.102
   Compiling autocfg v1.0.1
...
   Compiling duckscriptsdk v0.8.7
   Compiling ci_info v0.14.3
   Compiling cargo-make v0.35.0
    Finished release [optimized] target(s) in 1m 25s
   Replacing /usr/local/google/home/elango/.cargo/bin/cargo-make
   Replacing /usr/local/google/home/elango/.cargo/bin/makers
    Replaced package `cargo-make v0.33.0` with `cargo-make v0.35.0` (executables `cargo-make`, `makers`)
[cargo-make] INFO - Running Task: diplomat-get-rev
d98d31bdbdad3af98493ccfdc955b32fca886176 
[cargo-make] INFO - Build Done in 85.66 seconds.
```
And running the second time:
```
$ cargo make diplomat-get-rev
[cargo-make] INFO - cargo make 0.35.0
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: diplomat-get-rev
[cargo-make] INFO - Profile: development
[cargo-make] INFO - Running Task: legacy-migration
[cargo-make] INFO - Running Task: diplomat-get-rev
d98d31bdbdad3af98493ccfdc955b32fca886176 
[cargo-make] INFO - Build Done in 0.35 seconds.
```

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**